### PR TITLE
Add a test case for replication with short ttl

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #.
https://github.com/couchbaselabs/mobile-testkit/issues/1110

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
    1. SyncGateway Config with guest disabled = true and One user added (e.g. user1 / 1234)
    2. Create a new session on SGW for the user1 by using POST /_session.
       Capture the SyncGatewaySession cookie from the set-cookie in the response header.
    3. Start continuous push replicator on the LiteServ with SyncGatewaySession cookie.
       Make sure that the replicators start correctly
       Let the replicator run for more than 10% of the session expiry time specified when the session was created in step 2
    4. Cancel both push and pull replicator on the LiteServ
    5. Delete the session from SGW by sending DELETE /_sessions/ to SGW
    6. Repeat steps 2 through 6

### CODE REVIEW

File: testsuites/listener/shared/client_sg/test_replication.py [hunk 1]
```diff
@@ -1058,3 +1058,167 @@ def test_verify_open_revs_with_revs_limit_push_conflict(setup_client_syncgateway
...
+    log_info("sg_url: {}".format(sg_url))
+
+    client = MobileRestClient()
+    client.create_database(url=ls_url, name=ls_db)
+
+    # Get session header for user_1
+    cookie_name, session_id = client.create_session(url=sg_admin_url, db=sg_db, name="user_1", password="foo", ttl=10)
+
+    # session_header: SyncGatewaySession=a483be3248f740d810c09eb2c1b1f9198141bb15; Path=/db; Expires=Fri, 14 Apr 2017 03:54:46 GMT
+    session_header = "{}={}".format(cookie_name, session_id)
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_sg/test_replication.py]: add an assertion that the session header begins with `SyncGatewaySession=`

File: testsuites/listener/shared/client_sg/test_replication.py [hunk 2]
```diff
@@ -1058,3 +1058,167 @@ def test_verify_open_revs_with_revs_limit_push_conflict(setup_client_syncgateway
...
+        # Sanity test docs
+        ls_prefix = "ls_doc_" + str(attempts)
+        ls_docs = client.add_docs(url=ls_url, db=ls_db, number=num_docs_pushed, id_prefix=ls_prefix, channels=["ABC"])
+        assert len(ls_docs) == num_docs_pushed
+
+        attempts -= 1
+        # Sleep for 5 seconds after every add so as to expire the ttl
+        time.sleep(5)
+
+    all_docs = client.merge(ls_docs)
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_sg/test_replication.py]: What is the purpose of this?

File: testsuites/listener/shared/client_sg/test_replication.py [hunk 3]
```diff
@@ -1058,3 +1058,167 @@ def test_verify_open_revs_with_revs_limit_push_conflict(setup_client_syncgateway
...
+    assert len(replications) == 0, "All replications should be stopped"
+
+    # Delete the session
+    client.delete_session(url=sg_admin_url, db=sg_db, user_name="user_1", session_id=session_id)
+
+    # Get session header for user_1
+    cookie_name, session_id = client.create_session(url=sg_admin_url, db=sg_db, name="user_1", password="foo", ttl=10)
+
+    # session_header: SyncGatewaySession=a483be3248f740d810c09eb2c1b1f9198141bb15; Path=/db; Expires=Fri, 14 Apr 2017 03:54:46 GMT
+    session_header = "{}={}".format(cookie_name, session_id)
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_sg/test_replication.py]: Add a similar assert as above to make sure it received a session

**Comment by sethrosetter** [comment]
Can you add a comment where you expect it to blow up currently?

File: testsuites/listener/shared/client_sg/test_replication.py (hunk 1 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_sg/test_replication.py]: Done

File: testsuites/listener/shared/client_sg/test_replication.py (hunk 3 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_sg/test_replication.py]: Done

File: testsuites/listener/shared/client_sg/test_replication.py (hunk 2 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_sg/test_replication.py]: I was originally adding docs to both SG and CBL. I removed the code to add docs to SG but retained this piece of code. Removed it

**Pull request merged**